torch_xla/core/xla_model.py

```diff
@@ -616,6 +658,11 @@ def all_gather(value, dim=0, groups=None, output=None, pin_layout=False):
     A tensor which has, in the ``dim`` dimension, all the values from the
     participating replicas.
   """
+  if pin_layout == True and output == None:
```

Comment:
Since this difficulty is specific to TPU and GPU, should we spell out the condition here in the `if` test?

Reply:
That's fair, let me see if there is an easy way to determine the current device type.